File: IsarMainDisp/PythonVersiyonu/GenericISAR4predics.py
from IsarMain import IsarMain
from ParseObj import ParseObj
import figure
# from figure import figure1,figure2

# load rka file
import scipy.io as sio  # MATLAB 5.0 MAT-file

rkafile = sio.loadmat("rkafile.mat")
rkaArr = rkafile['a']

# rkafile = 0 # this is a 10000x11 array
# IsarMain
XX, YY, Th1, Ph1, ISAR_VVdB, ISAR_VHdB, ISAR_HVdB, ISAR_HHdB = IsarMain(rkaArr)

# wavefront object read
obj = ParseObj("../kepce.obj")

# plot 1)graph and 2)object from top(do later)
rd: int = 40
figure.figure(obj, XX, YY, rd, Th1, Ph1,
              ISAR_VVdB, ISAR_VHdB,ISAR_HVdB, ISAR_HHdB)

File: IsarMainDisp/GenericISAR4predics.m
%----------------------------------------------------------------
% This code for producing ISAR in predics
%----------------------------------------------------------------
clear ; close all; clc
%% ISAR(x,y)
% a=load('kepce.rka.out2');
% fname='kepce.obj';
% [XX,YY,Th1,Ph1,ISAR_VVdB,ISAR_VHdB,ISAR_HVdB,ISAR_HHdB] = IsarMain(a);
Isar;
%% Parse the object
obj = ParseObj(fname); % sadece altýna nesnenin resmini yerleþtiriyor.
% obj = ParseObjCodeGen(fname); % sadece altýna nesnenin resmini yerleþtiriyor.
%% figure Bu kýsým hesaplamasý çok uzun sürüyor.
% figure1
rd = 40;
figure1(obj,XX,YY,rd,Th1,Ph1,ISAR_VVdB,ISAR_VHdB);

% figure2
figure2(obj,XX,YY,rd,Th1,Ph1,ISAR_HVdB,ISAR_HHdB);
